Doc/library/ctypes.rst

```diff
@@ -2360,6 +2369,14 @@ Data types
       valid.  This object is only exposed for debugging; never modify the
       contents of this dictionary.
 
+   .. attribute:: __pointer_type__
+
+      This attributes is a pointer type that was created by calling
+      :func:`POINTER` for corresponding ctypes data type. If ``POINTER`` was
+      not called then this attribute contains ``None``.
```

Comment:
This should be referred from the warning. Users might want the docstring as well.  The requirements would be:

* This is an (internal?) cache for the `PONTER()`.

* This initially contains `None`.

* A pointer type is stored here (only once) when initialized with the ctypes data type.

cc @junkmd

Reply:
Optional:

* Non-ctype types can immitate the attribute for `POINTER()`.

The proxy demonstrates this.